src/PhpParser/Node/BetterNodeFinder.php

```diff
@@ -166,26 +166,43 @@ public function hasInstancesOfInFunctionLikeScoped(
             $types = [$types];
         }
 
-        $isFoundNode = false;
+        return $this->findInstancesOfScoped((array) $functionLike->stmts, $types) !== [];
+    }
+
+    /**
+     * @api to be used
+     *
+     * @template T of Node
+     * @param Node[] $nodes
+     * @param class-string<T>|array<class-string<T>> $type
+     * @return T[]
+     */
+    public function findInstancesOfScoped(array $nodes, string|array $type): array
+    {
+        /** @var T[] $foundNodes */
+        $foundNodes = [];
+
+        $types = is_array($type) ? $type : [$type];
+
         $this->simpleCallableNodeTraverser->traverseNodesWithCallable(
-            (array) $functionLike->stmts,
-            static function (Node $subNode) use ($types, &$isFoundNode): ?int {
+            $nodes,
+            static function (Node $subNode) use ($types, &$foundNodes): ?int {
                 if ($subNode instanceof Class_ || $subNode instanceof Function_ || $subNode instanceof Closure) {
                     return NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN;
                 }
 
                 foreach ($types as $type) {
                     if ($subNode instanceof $type) {
-                        $isFoundNode = true;
-                        return NodeTraverser::STOP_TRAVERSAL;
```

Comment:
stop traversal is essential here to not looking more if found, probably make another method with `$stopOnFound` flag so we have same performance

Reply:
actually, there is already `findInstancesOfInFunctionLikeScoped`, if there is a `$nodes` needed, probably better to allow array there, with make another private method for re-use as both method:

https://github.com/rectorphp/rector-src/blob/c5940d2cc1adbe39945e0a0f9029823a883671e6/src/PhpParser/Node/BetterNodeFinder.php#L228

Reply:
My suggestion is, make it allow `FunctionLike` in array of nodes[0]:

1. Keep `hasInstancesOfInFunctionLikeScoped` as is for performance reason
2. Modify the method to allow single `$nodes`, as `FunctionLike` reuse


```php
    /**
     * @api to be used
     *
     * @template T of Node
     * @param Node[] $nodes
     * @param class-string<T>|array<class-string<T>> $type
     * @return T[]
     */
    public function findInstancesOfScoped(array $nodes, string|array $type): array
    {
        // here verify only pass single nodes as FunctionLike
        if (count($nodes) === 1 && ($nodes[0] instanceof ClassMethod || $nodes[0] instanceof Function_ || $nodes[0] instanceof Closure)) {
            $nodes = (array) $nodes[0]->stmts;
        }

        if (is_string($types)) {
            $types = [$types];
        }

        /** @var T[] $foundNodes */
        $foundNodes = [];

        $this->simpleCallableNodeTraverser->traverseNodesWithCallable(
            $nodes,
            static function (Node $subNode) use ($types, &$foundNodes): ?int {
                if ($subNode instanceof Class_ || $subNode instanceof Function_ || $subNode instanceof Closure) {
                    return NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN;
                }

                foreach ($types as $type) {
                    if ($subNode instanceof $type) {
                        $foundNodes[] = $subNode;
                        return null;
                    }
                }

                return null;
            }
        );

        return $foundNodes;
```

3. Then, in `findInstancesOfInFunctionLikeScoped`, change to just call it:

```php
    public function findInstancesOfInFunctionLikeScoped(
        ClassMethod | Function_ | Closure $functionLike,
        string|array $types
    ): array {
        return $this->findInstancesOfScoped([$functionLike], $types);
    }
```

I will create an alternative PR for it.

Reply:
Sounds good 👍 

One of methods should return "all", the other only "first" for performance reasons.

Reply:
here the alternative PR:

- https://github.com/rectorphp/rector-src/pull/6324